odxtools/parameters/tablekeyparameter.py

```diff
@@ -19,6 +20,41 @@ def __init__(self, *, odx_id, table_ref, table_snref, table_row_snref, table_row
         self.table_snref = table_snref
         self.table_row_snref = table_row_snref
 
+        if self.table_ref is None and self.table_snref is None and \
+           self.table_row_ref is None and self.table_row_snref is None:
+            raise ValueError("Either a table or a table row must be defined.")
+
+    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
+        result = super()._build_odxlinks()
+
+        result[self.odx_id] = self
+
+        return result
+
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
+        super()._resolve_odxlinks(odxlinks)
+
+        if self.table_ref:
+            self._table = odxlinks.resolve(self.table_ref)
+        if self.table_row_ref:
+            self._table_row = odxlinks.resolve(self.table_row_ref)
+
+    def _resolve_snrefs(self, parent_dl: "DiagLayer") -> None:
+        super()._resolve_snrefs(parent_dl)
+
+        if self.table_snref:
+            self._table = parent_dl.local_diag_data_dictionary_spec.tables[self.table_snref]
```

Comment:
Only the local ones?

Reply:
I hope this corresponds to what's currently in `main`. doing this properly requires refactoring the value inheritance mechanism of diaglayers. (PR for this is in the pipeline, but doing this too here would be way too much IMO.)